src/webapi/date.rs

```diff
@@ -17,10 +121,703 @@ impl Date {
+    /// characters long (YYYY-MM-DDTHH:mm:ss.sssZ or ±YYYYYY-MM-DDTHH:mm:ss.sssZ, respectively). The timezone is always zero
+    /// UTC offset, as denoted by the suffix "Z".
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date.prototype.toisostring
+    #[inline]
+    pub fn to_iso_string(&self) -> String {
+        js!(
+            return @{self}.toISOString();
+        ).try_into().unwrap()
+    }
+
+    /// The toJSON() method returns a string representation of the Date object.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date.prototype.tojson
+    #[inline]
+    pub fn to_json(&self) -> String {
+        js!(
+            return @{self}.toJSON();
+        ).try_into().unwrap()
+    }
+
+    /// The toString() method returns a string representing the specified Date object.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toString)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date.prototype.tostring
+    #[inline]
+    pub fn to_string(&self) -> String {
+        js!(
+            return @{self}.toString();
+        ).try_into().unwrap()
+    }
+
+    /// The toTimeString() method returns the time portion of a Date object in human readable form in American English.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toTimeString)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date.prototype.totimestring
+    #[inline]
+    pub fn to_time_string(&self) -> String {
+        js!(
+            return @{self}.toTimeString();
+        ).try_into().unwrap()
+    }
+
+    /// The toUTCString() method converts a date to a string, using the UTC time zone.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toUTCString)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date.prototype.toutcstring
+    #[inline]
+    pub fn to_utc_string(&self) -> String {
+        js!(
+            return @{self}.toUTCString();
+        ).try_into().unwrap()
+    }
+
+    /// The valueOf() method returns the primitive value of a Date object.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/valueOf)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date.prototype.valueof
+    pub fn value_of(&self) -> f64 {
+        js!(
+            return @{self}.valueOf();
+        ).try_into().unwrap()
+    }
 }
 
 #[test]
 fn test_date_now() {
     let now = Date::now();
     assert!( now > 0.0 );
 }
+
+#[test]
+fn test_date_utc() {
```

Comment:
Can you put these tests in their own module? (Mostly to be consistent with the rest of the tests we have.)

```
#[cfg(test)]
mod tests {
...
}
```